### PR TITLE
Mtn v3 - change the method of generating a sequence of zones.

### PR DESCRIPTION
**Pull request by friendlyted**
Method "generate_with_tags_and_weights" created
Method is integrated into "init_terrain"

### CODE REVIEW

File: maps/mountain_fortress_v3/terrain.lua [hunk 1]
```diff
@@ -146,6 +146,78 @@ local function shuffle(tbl)
...
+            for _, i2 in pairs(arr2) do
+                if i1 == i2 then
+                    table.insert(result, i1)
+                end
+            end
+        end
+        return result
+    end
+
+    function next_value_provider()
```
Comment (Gerkiz) [inline maps/mountain_fortress_v3/terrain.lua]: Why aren't we utilizing the local variable here? The workflow is not happy :}

Comment (friendlyted) [reply maps/mountain_fortress_v3/terrain.lua]: **next_value_provider** is actually a variable, it is created within the **generate_with_tags_and_weights** function.
What exactly is wrong? I seriously don't get it :) We create inner function, return it and call from the "client" function (init_terrain).
**next_value_provider** returns new value each time, but **prev_tags** also keep tags of previous zone. 
If you mean we should pass prev tags throught the parameter... yeah, but in this case we are moving part of our logics outside, the incapsulation is breaking :(
I didn't find classes in lua, so have to do things like this.

Comment (friendlyted) [reply maps/mountain_fortress_v3/terrain.lua]: Sorry for my english :sweat_smile:

Comment (friendlyted) [reply maps/mountain_fortress_v3/terrain.lua]: Oh, if you mean the **intersect** function, it's an utility, so I couldn't decide where to put it, so I nested it

Comment (friendlyted) [reply maps/mountain_fortress_v3/terrain.lua]: I guess I got it. You mean the "local" one. Sorry, my first time with lua :)

**Comment by friendlyted** [comment]
@Gerkiz please check the **zones** variable for the correct tags. I'm not sure about them.

**Pull request merged**